### ISSUE TITLE
Cockroach with encryption-at-rest doesn't start on windows

### ISSUE THREAD

**Issue by aliher1911** [cockroachdb/cockroach#70700]
Validation script:
https://github.com/cockroachdb/bincheck/blob/master/bincheck

start-single-node fails:

```
I210924 17:03:53.526660 1 util/log/file_sync_buffer.go:238 ⋮ [config]   file created at: 2021/09/24 17:03:53
I210924 17:03:53.526703 1 util/log/file_sync_buffer.go:238 ⋮ [config]   running on machine: ‹DESKTOP-4F1N7FS›
I210924 17:03:53.526717 1 util/log/file_sync_buffer.go:238 ⋮ [config]   binary: CockroachDB CCL v21.2.0-beta.1 (x86_64-w64-mingw32, built 2021/09/24 15:27:21, go1.16.6)
I210924 17:03:53.526729 1 util/log/file_sync_buffer.go:238 ⋮ [config]   arguments: [‹cockroach.exe› ‹start-single-node› ‹--listening-url-file=cockroach-url› ‹--enterprise-encryption=path=cockroach-data,key=aes-128.key,old-key=plain›]
I210924 17:03:53.526762 1 util/log/file_sync_buffer.go:238 ⋮ [config]   log format (utf8=✓): crdb-v2
I210924 17:03:53.526772 1 util/log/file_sync_buffer.go:238 ⋮ [config]   line format: [IWEF]yymmdd hh:mm:ss.uuuuuu goid [chan@]file:line redactionmark \[tags\] [counter] msg
I210924 17:03:53.525562 1 util/log/flags.go:190  [-] 1  stderr capture started
panic: rename C:\Users\figar\Downloads\cockroach-v21.2.0-beta.1.windows-6.2-amd64\cockroach-v21.2.0-beta.1.windows-6.2-amd64\cockroach-data\COCKROACHDB_ENCRYPTION_REGISTRY.crdbtmp C:\Users\figar\Downloads\cockroach-v21.2.0-beta.1.windows-6.2-amd64\cockroach-v21.2.0-beta.1.windows-6.2-amd64\cockroach
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x41e292d]

goroutine 61 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl.(*DataKeyManager).rotateDataKeyAndWrite.func1(0xc0017dfa20, 0x8b42aa0, 0xc000068118, 0xc000fda660)
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl/pebble_key_manager.go:379 +0xcd
panic(0x4e94f20, 0xc000f52270)
        /usr/local/go/src/runtime/panic.go:965 +0x1c7
github.com/cockroachdb/cockroach/pkg/storage.(*PebbleFileRegistry).processBatchLocked(0xc000dea5b0, 0xc000f52198, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_file_registry.go:449 +0x1a8
github.com/cockroachdb/cockroach/pkg/storage.(*PebbleFileRegistry).SetFileEntry(0xc000dea5b0, 0xc00034ac5e, 0x1d, 0xc0001304c0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_file_registry.go:328 +0x20c
github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl.(*encryptedFS).Create(0xc000cb7fa0, 0xc00034abe0, 0x9b, 0x52a621e, 0x8, 0xc00034abe0, 0x9b)
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl/encrypted_fs.go:141 +0x408
github.com/cockroachdb/cockroach/pkg/storage.SafeWriteToFile(0x8c85fb0, 0xc000cb7fa0, 0xc000358f80, 0x7d, 0xc00034aaa0, 0x93, 0xc001018000, 0x28d, 0x28d, 0x0, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/file_util.go:31 +0xc5
github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl.(*DataKeyManager).rotateDataKeyAndWrite(0xc000fda660, 0x8b42aa0, 0xc000068118, 0xc001676510, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl/pebble_key_manager.go:394 +0x1ce
github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl.(*DataKeyManager).SetActiveStoreKeyInfo(0xc000fda660, 0x8b42aa0, 0xc000068118, 0xc0015980a0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl/pebble_key_manager.go:289 +0x2b7
github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl.newEncryptedEnv(0x8c869c8, 0xc000cb7e00, 0xc000dea5b0, 0xc000358f80, 0x7d, 0x0, 0xc0018be510, 0x8a, 0x8a, 0x3, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/engineccl/encrypted_fs.go:354 +0x409
github.com/cockroachdb/cockroach/pkg/storage.ResolveEncryptedEnvOptions(0xc0017e00d0, 0x3, 0x3, 0xc00034a320, 0x99, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:556 +0x1a6
github.com/cockroachdb/cockroach/pkg/storage.NewPebble(0x8b42b10, 0xc0016760f0, 0x0, 0x0, 0x0, 0xc000358f80, 0x7d, 0x0, 0x0, 0x40000000, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:595 +0x2b4
github.com/cockroachdb/cockroach/pkg/server.(*Config).CreateEngines(0xc0018f6a00, 0x8b42b10, 0xc0016760f0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/config.go:578 +0x1778
github.com/cockroachdb/cockroach/pkg/server.NewServer(0xc00082c000, 0xc000ac4700, 0xc000cd9580, 0x0, 0x0, 0x0, 0x0, 0x1dcd6500, 0xc0018770e0, 0x91, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:292 +0x32f
github.com/cockroachdb/cockroach/pkg/cli.runStart.func4.2(0xc00184df00, 0xc0003873e8, 0xc0019910e0, 0x8b42b10, 0xc0019d73b0, 0x1, 0x1f3ad364, 0xed8dffbf9, 0x0, 0x0, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:623 +0x68
github.com/cockroachdb/cockroach/pkg/cli.runStart.func4(0xc0003873e8, 0x8b42b10, 0xc0019d73b0, 0xc000cd95c0, 0xc00184df00, 0xc0019910e0, 0x1, 0x1f3ad364, 0xed8dffbf9, 0x0, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:762 +0x131
created by github.com/cockroachdb/cockroach/pkg/cli.runStart
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:594 +0x9f7
```

Bincheck run:
https://github.com/cockroachdb/bincheck/runs/3702093544?check_suite_focus=true

**Environment:**
 - CockroachDB version 21.2.0-beta.1
 - Server OS: Windows 10
 - Client app: bincheck

**Comment by jbowens** [comment]
I believe this is #69797

**Comment by aliher1911** [comment]
We should just close it then I think. As it is nothing new. We didn't realize that it would break release checks.

**Comment by tbg** [comment]
Why is this labeled as release-blocker? I can see why it should be a GA-blocker, but Serverless doesn't run on Windows (nor do any of our customers.. right?)

**Comment by jbowens** [comment]
@tbg This should just be closed. The issue is resolved on the release-21.2 branch and the original issue tracking this problem was labeled GA-Blocker.
